### ISSUE TITLE
Documentation page for `np.dtype` is malformed

### ISSUE THREAD

**Issue by eric-wieser** [numpy/numpy#17328]
Looking at https://numpy.org/devdocs/reference/generated/numpy.dtype.html#numpy.dtype, I notice:

* The docstring for `np.dtype` is not shown, even though `help(np.dtype)` shows it
* The `type` attribute is missing its docstring, even though `help(np.dtype.type)` shows it
* (minor) Attributes are shown as an indented block, methods are shown as a table

**Comment by Palash-Vishnani** [comment]
Can I work on the docstring of `np.dtype` if this issue is still open?

**Comment by melissawm** [comment]
Yes @Palash-Vishnani - there is no need to ask for permission to work on issues, feel free to go ahead :)

**Comment by Palash-Vishnani** [comment]
@eric-wieser here's the answer to your issue :

- This docstring of `np.dtype` is not shown because there is actually no docstring of `np.dtype`. I got to know this when I used `print(np.dtype.__doc__)` and it showed # None as the result, and when I used `.__doc__` attributes with various other classes of numpy it showed me the proper docstring. Talking about `help(np.dtype)`, it doesn't shows the docstring of dtype class, but the whole overview of that class.

- Same with `type` attribute, use `py.dtype.type.__doc__` and it will also return None as result. help() function will just show you the overview of its class not the docstring, because `type` attribute has no docstring.


![typeissue](https://user-images.githubusercontent.com/81798761/124246164-4bbc0280-db3e-11eb-8f65-0b104ff8e6f9.PNG)


I am currently working on there docstrings.

**Comment by charris** [comment]
Related to #18775, #18740?
